src/util.c

```diff
@@ -633,3 +633,62 @@ All facets connected.  No further nearby check necessary.\n");
     stl_verify_neighbors(stl);
   }
 }
+
+stl_file *stl_copy(stl_file *dst, const stl_file *src){
+
+  if(src->error) return NULL;
+
+  dst->fp               = NULL;
+  dst->facet_start      = NULL;
+  dst->edge_start       = NULL;       // edge_start is never allocated nor populated.
+  dst->heads            = NULL;       // Allocated and used only by stl_fill_holes and immediately freed.
+  dst->tail             = NULL;       // Same as ->heads.
+  dst->M                = src->M;
+  dst->neighbors_start  = NULL;
+  dst->v_indices        = NULL;
+  dst->v_shared         = NULL;
+  dst->stats            = src->stats;
+  dst->stats.type       = inmemory;
+  dst->error            = src->error;
+
+  if(src->facet_start != NULL){
+    
+    dst->facet_start = (stl_facet*)calloc(dst->stats.number_of_facets, sizeof(stl_facet));
+    if(dst->facet_start == NULL) perror("stl_copy");
+
+    for(int i = 0; i < dst->stats.number_of_facets; i++){
+      dst->facet_start[i] = src->facet_start[i];
+    }  
```

Comment:
Is there some kind of memcopy we could use to make this faster?

Reply:
Fixed